client/go/outline/client.go

```diff
@@ -42,67 +50,33 @@ type NewClientResult struct {
 
 // NewClient creates a new Outline client from a configuration string.
 func NewClient(transportConfig string) *NewClientResult {
-	client, err := newClientWithBaseDialers(transportConfig, net.Dialer{KeepAlive: -1}, net.Dialer{})
-	return &NewClientResult{
-		Client: client,
-		Error:  platerrors.ToPlatformError(err),
-	}
-}
-
-func newClientWithBaseDialers(transportConfig string, tcpDialer, udpDialer net.Dialer) (*Client, error) {
-	conf, err := parseConfigFromJSON(transportConfig)
-	if err != nil {
-		return nil, err
-	}
-	prefixBytes, err := ParseConfigPrefixFromString(conf.Prefix)
+	tcpDialer := transport.TCPDialer{Dialer: net.Dialer{KeepAlive: -1}}
+	udpDialer := transport.UDPDialer{}
+	client, err := newClientWithBaseDialers(transportConfig, &tcpDialer, &udpDialer)
 	if err != nil {
-		return nil, err
+		return &NewClientResult{Error: platerrors.ToPlatformError(err)}
 	}
-
-	return newShadowsocksClient(conf.Host, int(conf.Port), conf.Method, conf.Password, prefixBytes, tcpDialer, udpDialer)
+	return &NewClientResult{Client: client}
 }
 
-func newShadowsocksClient(
-	host string, port int, cipherName, password string, prefix []byte, tcpDialer, udpDialer net.Dialer,
-) (*Client, error) {
-	if err := validateConfig(host, port, cipherName, password); err != nil {
-		return nil, err
-	}
-
-	// TODO: consider using net.LookupIP to get a list of IPs, and add logic for optimal selection.
-	proxyAddress := net.JoinHostPort(host, fmt.Sprint(port))
-
-	cryptoKey, err := shadowsocks.NewEncryptionKey(cipherName, password)
+func newClientWithBaseDialers(transportConfig string, tcpDialer transport.StreamDialer, udpDialer transport.PacketDialer) (*Client, error) {
+	transportYAML, err := config.ParseConfigYAML(transportConfig)
```

Comment:
Are we still accepting the old JSON format here? Existing dynamic keys are still using JSON.

Reply:
Yes. YAML parses JSON. See the client_test.go